services/server/server.go

```diff
@@ -75,7 +76,22 @@ func CreateTopLevelDirectories(config *srvconfig.Config) error {
 		return err
 	}
 
-	return sys.MkdirAllWithACL(config.State, 0711)
+	if err := sys.MkdirAllWithACL(config.State, 0711); err != nil {
+		return err
+	}
+
+	if config.TempDir != "" {
+		if err := sys.MkdirAllWithACL(config.TempDir, 0711); err != nil {
+			return err
+		}
+		if runtime.GOOS == "windows" {
+			os.Setenv("TEMP", config.TempDir)
+			os.Setenv("TMP", config.TempDir)
```

Comment:
Could it be only `TMP`?

https://pkg.go.dev/os#TempDir

> On Windows, it uses GetTempPath, returning the first non-empty value from %TMP%, %TEMP%, %USERPROFILE%, or the Windows directory.

Reply:
So unfortunately, this is not "golang" using the temp dir (although golang would also see it from this location). The system service VMCompute inspects the TMP/TEMP settings of the calling process and uses its value when extracting the Windows layer. So we need to have them overridden so that the System Environment Variables both point to our custom location.

Reply:
That line deserves this comment :)

Reply:
LOL. Ok will push a new commit. :)